version: "3.9"
services:
  db:
    image: mysql
    command: --default-authentication-plugin=mysql_native_password
    environment:
      MYSQL_ROOT_PASSWORD: $DB_PASS
      MYSQL_DATABASE: $DB_DATABASE
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      timeout: 20s
      retries: 10
    volumes:
      - db:/var/lib/mysql
  app:
    image: luisdonaldo/docker
    depends_on:
      db:
        condition: service_healthy
    environment:
      APP_DB_URL: jdbc:mysql://db:$DB_DOCKER_PORT/$DB_DATABASE?useSSL=false
      APP_DB_DRIVER_CLASSNAME: com.mysql.jdbc.Driver
      APP_DB_PLATFORM: "org.hibernate.dialect.MySQL5InnoDBDialect"
      APP_DB_USERNAME: $DB_USER
      APP_DB_PASSWORD: $DB_PASS
    ports:
      - $HOST_APP_PORT:8080

  phpmyadmin:
    image: phpmyadmin

    ports:
      - $HOST_DB_WEB_CLIENT_PORT:80
    environment:
      PMA_HOST: db
      PMA_PORT: $DB_DOCKER_PORT
      PMA_ARBITRARY: 1

volumes:
  db: